### PR TITLE
Favorites

### PR DESCRIPTION
**Pull request by farosch**
Implemented a favorites menu in the connection tree window.

## Description
- Added a new drop down menu in the conenction tree window
- Introduced new connection property
- Introduced new inherited connection property

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: mRemoteV1/Config/Serializers/ConnectionSerializers/MsSql/DataTableSerializer.cs [hunk 1]
```diff
@@ -108,6 +108,7 @@ private void CreateSchema(DataTable dataTable)
             dataTable.Columns.Add("PostExtApp", typeof(string));
             dataTable.Columns.Add("MacAddress", typeof(string));
             dataTable.Columns.Add("UserField", typeof(string));
+            dataTable.Columns.Add("Favorite", typeof(bool));
```
Comment (sparerd) [inline mRemoteV1/Config/Serializers/ConnectionSerializers/MsSql/DataTableSerializer.cs]: Should connection favorites be shared among all users or specific to each user? If we both share a SQL DB connection list and I set connection "app-001.example.com" as a favorite, should you also see that as a favorite? I could see reasons for wanting this to be a personal or shared setting. 

If you think that sharing the favorites is important, you can leave this as it is.
If you think it should be a personal setting, revert the current changes to the database/schema files and instead update the following:
- `LocalConnectionPropertiesModel`
- `SqlConnectionsLoader.ApplyLocalConnectionProperties(ContainerInfo rootNode)`
- `SqlConnectionsSaver.PropertyIsLocalOnly(string property)`
- `SqlConnectionsSaver.UpdateLocalConnectionProperties(ContainerInfo rootNode)`

(Being forced to type out the steps to add a single property makes me realize we need to refactor the local connection properties code. It's ridiculous that we need to touch 4 completely separate places for 1 change.)

**Comment by sparerd** [comment]
Got an exception when trying to set the `Favorite` setting of a connection.

```
Exception Message
"Object reference not set to an instance of an object."

Stack trace
   at System.Windows.Forms.PropertyGridInternal.GridEntry.NonParentEquals(Object obj)
   at System.Windows.Forms.PropertyGridInternal.GridEntry.Equals(Object obj)
   at System.Windows.Forms.PropertyGridInternal.PropertyGridView.GetRowFromGridEntry(GridEntry gridEntry)
   at System.Windows.Forms.PropertyGridInternal.PropertyGridView.SelectGridEntry(GridEntry gridEntry, Boolean fPageIn)
   at System.Windows.Forms.PropertyGridInternal.PropertyGridView.OnGridEntryLabelClick(Object s, EventArgs e)
   at System.Windows.Forms.PropertyGridInternal.GridEntry.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.PropertyGridInternal.GridEntry.OnLabelClick(EventArgs e)
   at System.Windows.Forms.PropertyGridInternal.GridEntry.OnMouseClick(Int32 x, Int32 y, Int32 count, MouseButtons button)
   at System.Windows.Forms.PropertyGridInternal.PropertyDescriptorGridEntry.OnMouseClick(Int32 x, Int32 y, Int32 count, MouseButtons button)
   at System.Windows.Forms.PropertyGridInternal.PropertyGridView.OnMouseDown(MouseEventArgs me)
   at System.Windows.Forms.Control.WmMouseDown(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.PropertyGridInternal.PropertyGridView.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

image

File: mRemoteV1/Config/Serializers/ConnectionSerializers/MsSql/DataTableSerializer.cs (hunk 1 above)
Comment (farosch) [reply mRemoteV1/Config/Serializers/ConnectionSerializers/MsSql/DataTableSerializer.cs]: I'll turn the favorite property into a local property - makes much more sense for every user to be able to set favorites for himself

**Comment by farosch** [comment]
See #1320

**Pull request closed without merging**